crates/ruff_linter/src/rules/flake8_simplify/rules/open_file_with_context_handler.rs

```diff
@@ -112,32 +112,21 @@ fn match_exit_stack(semantic: &SemanticModel) -> bool {
     false
 }
 
-/// Return `true` if `func` is the builtin `open` or `pathlib.Path(...).open`.
+/// Return `true` if the expression is an `open` call or temporary file constructor.
 fn is_open(semantic: &SemanticModel, func: &Expr) -> bool {
-    // Ex) `open(...)`
-    if semantic.match_builtin_expr(func, "open") {
-        return true;
-    }
-
-    // Ex) `pathlib.Path(...).open()`
-    let Expr::Attribute(ast::ExprAttribute { attr, value, .. }) = func else {
-        return false;
-    };
-
-    if attr != "open" {
-        return false;
-    }
-
-    let Expr::Call(ast::ExprCall {
-        func: value_func, ..
-    }) = &**value
-    else {
+    let Some(qualified_name) = semantic.resolve_qualified_name(func) else {
         return false;
     };
```

Comment:
The change you've made to this function means that the rule no longer detects `pathlib.Path("foo").open("bar)"` calls, because the qualified name of `pathlib.Path("foo")` is not `Some(["pathlib", "Path"])`, it's `None`. The fully qualified name of `pathlib.Path` (an attribute expression) is `Some(["pathlib", "Path"])`, but `pathlib.Path("foo")` is a call expression. This is why a number of hits are going away in the ecosystem report, and also why one of the hits is going away in the snapshot.